app/assets/javascripts/angular/controllers/mobile/messages.js

```diff
@@ -0,0 +1,225 @@
+    }
+  ]);
+
+  angular.module('wtsApp').controller('messagesCtrl', ['$scope', '$http', '$interval', function($scope, $http, $interval) {
+    // Default state for Messages.
+    $scope.initializeMessages = function() {
+      $scope.resetMessages();
+
+      // Message Filtering
+      $scope.filterOptions = [
+        {"id": 0, "title": "Show all messsages"},
+        {"id": 1, "title": "Show those waiting on me"},
+        {"id": 2, "title": "Show those I am waiting on"}
+      ];
+      $scope.messageFilter = $scope.filterOptions[0];
+    }
+
+    // A team should not be able to send messages to itself.
+    $scope.validTeams = _.reject($scope.teams, function(team) {
+      return team.id === $scope.myCountryId;
+    });
+
+    // Reset the new message form.
+    $scope.resetNewMessage = function() {
+      $scope.newMessage = {};
+      $scope.message = {
+        sender: null,
+        content: null
+      };
+    }
+
+    // Selecting conversations / new message form.
+    $scope.selectConversation = function(conversation) {
+      $scope.resetNewMessage();
+      $scope.messageIsActive = conversation;
+      $scope.selectedMessage = conversation;
+    };
+
+    $scope.createNewMessage = function() {
+      $scope.resetNewMessage();
+      $scope.messageIsActive = null;
+      $scope.selectedMessage = 'new';
+    };
+
+    $scope.resetMessages = function() {
+      $scope.resetNewMessage();
+      $scope.messageIsActive = null;
+      $scope.selectedMessage = null;
+    };
+
+    // Fire the request to send a message.
+    $scope.sendMessage = function(message) {
+      // Set us as being the one who sent the message.
+      message.sender = $scope.myCountryId;
+
+      // Error handling
+      $scope.messageError = null;
+      if (message.content === '' || message.content === null) {
+        $scope.messageError = "You must enter a message";
+        return;
+      } else if (message.recipient === null) {
+        $scope.messageError = "You must choose a team";
+        return;
+      }
+
+      // Create the new Message object.
+      $scope.newMessage = { "message": {} }
+      $scope.newMessage.message = angular.copy(message);
+      $scope.newMessage.message.recipient = parseInt(message.recipient);
+
+      $http.post('/api/messages', $scope.newMessage).then(
+      function successCalback(response) {
+        $scope.pushNewMessage(response.data, message.recipient);
+        $scope.resetNewMessage();
+      },
+      function errorCallback(response) {
+
+      });
+    }
+
+    // After submitting a new Message, do some updating to the view.
+    $scope.pushNewMessage = function(savedMessage, conversationPartnerId) {
+      // If there is a conversation already with this partner, grab it.
+      var conversationToUpdate = _.filter($scope.messages, function(message) {
+        return message.conversation_partner.id === conversationPartnerId;
```

Comment:
There's a question I have out to Noam of whether we should even display existing teams in the "New Message" dropdown because it's going to add it to existing conversations no matter what, which might seem strange. It's currently bad UX in either case.